apikey-service/src/main/java/eu/europeana/apikey/controller/ApikeyController.java

```diff
@@ -386,7 +358,22 @@ public ApikeyController(ApikeyRepo apikeyRepo) {
         this.apikeyRepo.save(apikey);
 
         // Welcome, gringo!
-        return new ResponseEntity<>(headers, HttpStatus.NO_CONTENT);
+        return new ResponseEntity<>(HttpStatus.NO_CONTENT);
+    }
+
+    private String getApikey(HttpServletRequest httpServletRequest) {
+        String authorization = httpServletRequest.getHeader("Authorization");
+        if (authorization != null) {
+            String patternString = APIKEY_PATTERN;
+
+            Pattern pattern = Pattern.compile(patternString);
+            Matcher matcher = pattern.matcher(authorization);
+
+            if (matcher.find()) {
+                return matcher.group(1);
```

Comment:
shoudn't group() be sufficient? the group() methods throw runtime exceptions that are not handled ...

Reply:
No, group() is not sufficient, it calls group(0) while here is group(1).